### PR TITLE
getCustomDataForAnnotation

### PR DESCRIPTION
**Pull request by dcupidon**
Exposed an API that retrieves an annotation's customData property.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -1919,6 +1919,30 @@ this._viewer.getAnnotationListOnPage(2).then((annotations) => {
...
+Returns a Promise.
+
+Promise Parameters: 
+Name | Type | Description
+--- | --- | ---
+value | string | the `customData` property associated with the given key
+
+```js
+this._viewer.getCustomDataForAnnotation("Pdftron", 2, "version").then((value) => {
+  console.log('The PDFNet version is', value);
```
Comment (JamieDass) [inline API.md:1942]: This sample snippet log output is a bit confusing, seems to be implying that the custom data is used to store the PDFNet version. Maybe something like
`console.log('The custom data for the key "version" is', value);`
(also not a great example… open to other ideas)

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 2]
```diff
@@ -2723,6 +2723,35 @@ public WritableArray getAnnotationListOnPage(int pageNumber) throws PDFNetExcept
...
+
+        String customData = "";
+
+        if (pdfViewCtrl != null && pdfDoc != null) {
+            boolean shouldUnlockRead = false;
+            try {
+                pdfDoc.lockRead();
+                shouldUnlockRead = true;
+                ArrayList<Annot> annots = pdfViewCtrl.getAnnotationsOnPage(pageNumber);
+                for (Annot annot : annots) {
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2738]: - [ ] instead, use `Annot ann = ViewerUtils.getAnnotById(getPdfViewCtrl(), id, page);`

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 3]
```diff
@@ -2723,6 +2723,35 @@ public WritableArray getAnnotationListOnPage(int pageNumber) throws PDFNetExcept
...
+    public String getCustomDataForAnnotation(String annotationID, int pageNumber, String key) throws PDFNetException {
+        PDFViewCtrl pdfViewCtrl = getPdfViewCtrl();
+        PDFDoc pdfDoc = getPdfDoc();
+
+        String customData = "";
+
+        if (pdfViewCtrl != null && pdfDoc != null) {
+            boolean shouldUnlockRead = false;
+            try {
+                pdfDoc.lockRead();
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2735]: - [ ] lock on PDFViewCtrl instead

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: API.md (hunk 1 above)
Comment (dcupidon) [reply API.md:1942]: I see what you're saying. Would having a call to `setPropertiesForAnnotation` where the custom data is set before `getCustomDataForAnnotation` be better?

File: API.md [hunk 4]
```diff
@@ -1938,8 +1938,14 @@ Name | Type | Description
...
 ```js
-this._viewer.getCustomDataForAnnotation("Pdftron", 2, "version").then((value) => {
-  console.log('The PDFNet version is', value);
+this._viewer.setPropertiesForAnnotation("annotation1", 2, {
+  customData: {
+    data: "Nice annotation"
+  }
+}).then(() => {
+  this._viewer.getCustomDataForAnnotation("annotation1", 2, "data").then((value) => {
+    console.log(value == "Nice annotation");
```
Comment (sgong-pdftron) [inline API.md:1947]: javascript equal should use 3 equals

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 5]
```diff
@@ -2723,6 +2723,33 @@ public WritableArray getAnnotationListOnPage(int pageNumber) throws PDFNetExcept
         return annotations;
     }
 
+    public String getCustomDataForAnnotation(String annotationID, int pageNumber, String key) throws PDFNetException {
+        PDFViewCtrl pdfViewCtrl = getPdfViewCtrl();
+        PDFDoc pdfDoc = getPdfDoc();
+
+        String customData = "";
+
+        if (pdfViewCtrl != null && pdfDoc != null) {
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2732]: - [ ] pdfdoc is not used, can remove

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

**Review by sgong-pdftron — APPROVED** [review]

**Pull request merged**